### ISSUE TITLE
SNOW-1292755: SQL compilation error when using a role containing hyphens

### ISSUE THREAD

**Issue by dijitali** [snowflakedb/snowflake-cli#954]
### SnowCLI version

`2.1.2`

### Python version

`Python 3.12.2`

### Platform

`macOS-14.4.1-arm64-arm-64bit`

### What happened

Connecting with a `role` containing hyphens gives error

### Console output

```Shell
$ snow connection test --connection example
╭─ Error ────────────────────────────────────────────────────────────────────────────────────────╮
│ 001003 (42000): 01b362cf-3202-76a5-0000-44b5026d4166: SQL compilation error:                   │
│ syntax error line 1 at position 13 unexpected '-'.                                             │
╰────────────────────────────────────────────────────────────────────────────────────────────────╯
```

And in Query History, I can see:

![image](https://github.com/snowflakedb/snowflake-cli/assets/4665045/df6e4456-b13b-4d7b-8972-98895567a223)

### How to reproduce

Add a connection that uses a role containing hyphens (or if the default role contains hyphens):

```bash
snow connection add -n example \
  -a example \
  -h xyz.snowflakecomputing.com \
  -u user@example.com \
  -w USERS \
  -r EXAMPLE-USERS \
  -s EXAMPLE_SCHEMA \
  -a externalbrowser
```

**Comment by dijitali** [comment]
NB: have also tried directly modifying the `~/Library/Application\ Support/snowflake/config.toml` file to use double-quoted identifiers per: [Identifier requirements - Double-quoted identifiers](https://docs.snowflake.com/en/sql-reference/identifiers-syntax#double-quoted-identifiers) but putting `role = ""EXAMPLE-USERS""` in the `config.toml` gives:

```text
An unknown error happened while loading '~/Library/Application Support/snowflake/config.toml'
```

**Comment by dijitali** [comment]
@sfc-gh-turbaszek - that gives the same SQL error unfortunately:

```text
╭─ Error ──────────────────────────────────────────────────────────────────────────────────────╮
│ 001003 (42000): 01b368d0-3202-7739-0000-44b5026dba4e: SQL compilation error:                 │
│ syntax error line 1 at position 13 unexpected '-'.                                           │
╰──────────────────────────────────────────────────────────────────────────────────────────────╯
```

Also just to note the real value I'm using for role is `CIGA-PRN-USERS`, or in the TOML:

```toml
[connections.example]
account = "REMOVED"
user = "REMOVED"
host = "REMOVED.snowflakecomputing.com"
database = "REMOVED"
schema = "REMOVED"
role = "\"CIGA-PRN-USERS\""
warehouse = "REMOVED"
authenticator = "externalbrowser"
```

**Comment by sfc-gh-pczajka** [comment]
We have prepared a fix, it will be released in 2.2.0

**Comment by dijitali** [comment]
Thanks!  I assume that's #971?  So I can try it out now if I install from source?
